From 680ade3e79803cf06004f6e2b5060170ee8eeca0 Mon Sep 17 00:00:00 2001
From: John Linn <john.linn@xilinx.com>
Date: Tue, 31 Jul 2012 16:20:46 -0700
Subject: [PATCH 593/628] Xilinx: ARM: MTD: Clean up of QSPI dual support

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 7dae15f1ba7a8d77f9ba5c9637bd2e2d1222ecaa

The dual support was based on kernel configuration
which forced unique kernels for the 702 and 706
boards. Now it's only based on the device tree.

Signed-off-by: John Linn <john.linn@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 arch/arm/boot/dts/zynq-zc706.dts |   57 ++++++++++++++++++++------------------
 1 files changed, 30 insertions(+), 27 deletions(-)

diff --git a/arch/arm/boot/dts/zynq-zc706.dts b/arch/arm/boot/dts/zynq-zc706.dts
index adc8ee5..d4b36de 100644
--- a/arch/arm/boot/dts/zynq-zc706.dts
+++ b/arch/arm/boot/dts/zynq-zc706.dts
@@ -1,5 +1,4 @@
 /dts-v1/;
-
 / {
 	model = "Xilinx Zynq ZC706";
 	compatible = "xlnx,zynq-zc706";
@@ -138,38 +137,42 @@
 			#size-cells = <0>;
 			is-dual = <1>;
 			flash@0 {
-			compatible = "n25q128x2";
-			reg = <0x0>;
-			spi-max-frequency = <30000000>;
-			#address-cells = <1>;
-			#size-cells = <1>;
-			partition@qspi-fsbl {
-				label = "qspi-fsbl";
-				reg = <0x0 0x80000>;
+				compatible = "n25q128";
+				reg = <0x0>;
+				spi-max-frequency = <30000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				partition@qspi-fsbl {
+					label = "qspi-fsbl";
+					reg = <0x0 0x80000>;
+				};
+				partition@qspi-u-boot {
+					label = "qspi-u-boot";
+					reg = <0x80000 0x80000>;
 				};
-			partition@qspi-u-boot {
-				label = "qspi-u-boot";
-				reg = <0x80000 0x80000>;
+				partition@qspi-linux {
+					label = "qspi-linux";
+					reg = <0x100000 0x500000>;
 				};
-			partition@qspi-linux {
-				label = "qspi-linux";
-				reg = <0x100000 0x500000>;
+				partition@qspi-device-tree {
+					label = "qspi-device-tree";
+					reg = <0x600000 0x20000>;
 				};
-			partition@qspi-device-tree {
-				label = "qspi-device-tree";
-				reg = <0x600000 0x20000>;
+				partition@qspi-user {
+					label = "qspi-user";
+					reg = <0x620000 0xE0000>;
 				};
-			partition@qspi-user {
-				label = "qspi-user";
-				reg = <0x620000 0xE0000>;
+				partition@qspi-scratch {
+					label = "qspi-scratch";
+					reg = <0x700000 0x100000>;
 				};
-			partition@qspi-scratch {
-				label = "qspi-scratch";
-				reg = <0x700000 0x100000>;
+				partition@qspi-rootfs {
+					label = "qspi-rootfs";
+					reg = <0x800000 0x1000000>;
 				};
-			partition@qspi-rootfs {
-				label = "qspi-rootfs";
-				reg = <0x800000 0x800000>;
+				partition@qspi-test {
+					label = "qspi-test";
+					reg = <0x1800000 0x800000>;
 				};
 			};
 		};
-- 
1.7.5.4

